subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformInputArtifactIntegrationTest.groovy

```diff
@@ -1161,6 +1161,56 @@ class ArtifactTransformInputArtifactIntegrationTest extends AbstractDependencyRe
         annotation << ["Classpath", "CompileClasspath"]
     }
 
+    def "honors runtime classpath normalization for input artifact"() {
+        settingsFile << "include 'a', 'b', 'c'"
+        setupBuildWithColorTransformAction {
+            produceJars()
+        }
+        buildFile << """
+            project(':a') {
+                dependencies {
+                    implementation project(':b')
+                    implementation project(':c')
+                }
+                
+                normalization {
+                    runtimeClasspath {
+                        ignore("ignored.txt")
+                    }
+                }
+            }
+            
+            abstract class MakeGreen implements TransformAction {
+                @InputArtifact @Classpath
+                abstract File getInput()
+                
+                void transform(TransformOutputs outputs) {
+                    println "processing \${input.name}"
+                    def output = outputs.file(input.name + ".green")
+                    output.text = input.text + ".green"
+                }
+            }
+        """
+
+        when:
+        executer.withArguments("-PbEntryName=ignored.txt")
+        succeeds(":a:resolve")
+
+        then:
+        result.assertTasksNotSkipped(":b:producer", ":c:producer", ":a:resolve")
+        transformed("b.jar", "c.jar")
+        outputContains("result = [b.jar.green, c.jar.green]")
+
+        when:
+        executer.withArguments("-PbEntryName=ignored.txt", "-PbContent=different")
+        succeeds(":a:resolve")
+
+        then: // change is ignored due to normalization
+        result.assertTasksNotSkipped(":b:producer", ":a:resolve")
+        transformed()
+        outputContains("result = [b.jar.green, c.jar.green]")
```

Comment:
It is not apparent here why these assertions prove that the change was ignored. How can we make this more clear?

Reply:
`transformed()` asserts that no transformations have been executed. It seems like that isn't clear enough. What would you expect?

Reply:
I think having a method like `nothingTransformed()` or `noTransformExecuted()` would be better. Let's not hold up the PR for this, though.